### PR TITLE
pythonPackages.datasette-template-sql: init at 1.0.2

### PR DESCRIPTION
**Pull request by MostAwesomeDude**
###### Motivation for this change

Datasette, already packaged, can be augmented with plugins. This plugin is provided by upstream developers as a way to perform SQL queries while building templates which are formatting SQL queries.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129072` at 449f84a3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.datasette-template-sql</li>
    <li>python39Packages.datasette-template-sql</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/datasette/template-sql.nix:20:0:
```
   |
20 |     description = "Datasette plugin for executing SQL queries from templates";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/datasette/template-sql.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchFromGitHub {
+    owner = "simonw";
+    repo = pname;
+    rev = version;
+    sha256 = "1ag5f62y445jscxnklcfd84pjinkknmrpk1kmm2j121p7484hrsn";
+  };
+
+  propagatedBuildInputs = [ datasette ];
+
+  checkInputs = [ pytest pytest-asyncio sqlite-utils ];
```
Comment (fabaff) [inline pkgs/development/python-modules/datasette/template-sql.nix:17]: Looks like that we could use `pytestCheckHook`.

Please add `pythonImportsCheck` as a additional safe-guard.

Comment (MostAwesomeDude) [reply pkgs/development/python-modules/datasette/template-sql.nix]: Done, thanks.

File: pkgs/development/python-modules/datasette/template-sql.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "simonw";
+    repo = pname;
+    rev = version;
+    sha256 = "1ag5f62y445jscxnklcfd84pjinkknmrpk1kmm2j121p7484hrsn";
+  };
+
+  propagatedBuildInputs = [ datasette ];
+
+  checkInputs = [ pytestCheckHook pytest pytest-asyncio sqlite-utils ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/datasette/template-sql.nix:17]:
```suggestion
  checkInputs = [ pytestCheckHook pytest-asyncio sqlite-utils ];
```

File: pkgs/development/python-modules/datasette/template-sql.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ buildPythonPackage, fetchFromGitHub, lib,
+  datasette, pytestCheckHook, pytest, pytest-asyncio, sqlite-utils }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/datasette/template-sql.nix:2]:
```suggestion
{ buildPythonPackage
, fetchFromGitHub
, lib
, datasette
, pytestCheckHook
, pytest-asyncio
, sqlite-utils
}:
```

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -1815,6 +1815,8 @@ in {
 
   datasette = callPackage ../development/python-modules/datasette { };
 
+  datasette-template-sql = callPackage ../development/python-modules/datasette/template-sql.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:1818]:
```suggestion
  datasette-template-sql = callPackage ../development/python-modules/datasette-template-sql { };
```

either way is fine.

**Comment by MostAwesomeDude** [comment]
I accepted all suggestions, and fixed the build so it still works. I just tested that my prototype Datasette configuration, which relies on this plugin, still works too. Thanks for your patience and hard work.

**Pull request merged**